pkg/validator/promql_expression_helpers_test.go

```diff
@@ -65,8 +66,61 @@ func TestGetExpressionUsedLabelsForMetric(t *testing.T) {
 
 	for i, test := range tests {
 		t.Run(fmt.Sprintf("test_case_%d", i), func(t *testing.T) {
-			labels, err := getExpressionUsedLabelsForMetric(test.expr, regexp.MustCompile(test.metric))
-			assert.ElementsMatch(t, labels, test.expected, "Expected labels %v, but got %v", test.expected, labels)
+			l, err := getExpressionUsedLabelsForMetric(test.expr, regexp.MustCompile(test.metric))
+			assert.ElementsMatch(t, l, test.expected, "Expected labels %v, but got %v", test.expected, l)
+			if !errors.Is(err, test.expectedErr) {
+				t.Errorf("Expected error %v, but got %v", test.expectedErr, err)
+			}
+		})
+	}
+}
+
+func TestGetLabelMatchersForMetricRegexp(t *testing.T) {
+	MustNewMatcher := func(typ labels.MatchType, label, value string) *labels.Matcher {
+		matcher, err := labels.NewMatcher(typ, label, value)
+		if err != nil {
+			t.Fatalf("failed to create matcher: %v", err)
+		}
+		return matcher
+	}
+
+	MatchersToString := func(matchers []*labels.Matcher) []string {
+		result := make([]string, 0, len(matchers))
+		for _, m := range matchers {
+			result = append(result, m.String())
+		}
+		return result
+	}
+
+	tests := []struct {
+		expr         string
+		metricRegexp string
+		expected     []*labels.Matcher
+		expectedErr  error
+	}{
+		{expr: "up{bar='foo'}", metricRegexp: "kube_pod_labels", expected: []*labels.Matcher{}, expectedErr: nil},
+		{expr: "up{bar='foo'}", metricRegexp: "up", expected: []*labels.Matcher{
+			MustNewMatcher(labels.MatchEqual, "__name__", "up"), MustNewMatcher(labels.MatchEqual, "bar", "foo"),
+		}, expectedErr: nil},
+		{expr: "up{bar!='foo'}", metricRegexp: "up", expected: []*labels.Matcher{
```

Comment:
so the regexp functionality is actually checked
```suggestion
		{expr: "up{bar!='foo'}", metricRegexp: ".*", expected: []*labels.Matcher{
```